docs/design-documents/hal/0005-pin-names-Arduino-Uno-standard.md

```diff
@@ -0,0 +1,210 @@
+    #define ARDUINO_UNO_I2C_SCL ARDUINO_UNO_D15
+
+    // Arduino Uno SPI signals aliases
+    #define ARDUINO_UNO_SPI_CS   ARDUINO_UNO_D10
+    #define ARDUINO_UNO_SPI_MOSI ARDUINO_UNO_D11
+    #define ARDUINO_UNO_SPI_MISO ARDUINO_UNO_D12
+    #define ARDUINO_UNO_SPI_SCK  ARDUINO_UNO_D13
+
+    // Arduino Uno UART signals aliases
+    #define ARDUINO_UNO_UART_TX  ARDUINO_UNO_D0
+    #define ARDUINO_UNO_UART_RX  ARDUINO_UNO_D1
+
+    #endif // TARGET_FF_ARDUINO_UNO
+
+**Other pin functions**
+
+Some Dx pinnames may support the usage of PWM or Timers functions. Although this is recomended, it's not a mandatory as requirement to be compliant with the Arduino Uno standard for Mbed boards.
+
+Note this might be one of the main differencess accross Mbed boards and therefore the application should not assume the same behaviour for PWM and Timers for them.
+
+The RESET signal should be defined in the Arduino Uno connector to let the user or an external component put the MCU in reset state.
+
+The VIN signal isn't strictly defined in the Arduino standard and in some cases may work as a bi-directional power supply. Tipically it could accept between 5 to 12 volts, but it's not a requirement.
+
+### Usage of Arduino Uno pin-names in an application
+
+**Configuration**
+
+The pin names defined to each of the MCUs in PinNames.h are considered the default configuration for every target in Mbed OS.
+They are picked automatically at compile time and can used in the application.
+
+However, it's possible to override the default configuration. This is the list of configuration files and a description of each of them:
+
+- `PinNames.h`: includes the default configuration for Arduino Uno pins and how they are connected to the MCU as described in this document.
+- `mbed_lib.json`: configuration for specific components to use the default Arduino Uno pin-names or select other MCU pins if required. This file can be defined in Mbed OS or in an external library for the component, for [example](https://github.com/ARMmbed/mbed-os/blob/master/storage/blockdevice/COMPONENT_SD/mbed_lib.json).
+- `mbed_app.json`: application configuration that inherits the `mbed_lib.json` if exists and can override the pin configuration for the component if required. This file is usually placed at the root directory of the application.
+- main application: the `.cpp` application could initialize the drivers with the default Arduino Uno pin-name configution, use the configuration in `mbed_app.json`, or use custom pins if required.
+
+**Examples of usage of Arduino Uno pin names**
+
+This section provides a few examples with guidelines on using Arduino Uno pin names. It's not a guide to use Mbed Drivers API (please refer to the [documentation](https://os.mbed.com/docs/mbed-os/latest/apis/drivers.html) for details).
+
+It's possible to use digital signals in both Dx and Ax pins:
+
+    DigitalOut myGPIO1(ARDUINO_UNO_D3); // Output signal
+    DigitalOut myGPIO2(ARDUINO_UNO_D4); // Input signal
```

Comment:
should this not be  the following

 DigitalIn myGPIO2(ARDUINO_UNO_D4); // Input signal

Reply:
Good catch!